molecularnodes/io/parse/molecule.py

```diff
@@ -580,6 +591,16 @@ def att_sec_struct():
                 print(
                     f'Failed adding {att["name"]} after {time.process_time() - start} s')
 
+    if centre.lower() == 'mass':
```

Comment:
The issue there is the need for atom names to map to element names to map to masses. I can move the centre by CoM chunk of code to before the `bl.obj.create_object()` line but I'll also need to get the element and mass attributes first.

Reply:
I'm just noticing that the MDAnalysis guesser is only implemented for trajectories loaded in from local. For PDB and CIF files, the respective biotite.structure.io modules are used to load the structure into an atom array. This atom array has an `element` attribute for each atom. If the PDB file doesn't have values in the 76-78 column indices, then biotite has a guesser for elements at  https://github.com/biotite-dev/biotite/blob/123e5332bd78fe7189bd5cb6e3147742cf5d73fe/src/biotite/structure/io/general.py#L248-L271 .

Reply:
This guesser is certainly not bug free, especially when coarse-grained particles are considered. It checks atom names against a list of upper()'d element symbols. If the atom name matches an element symbol, then the element is assigned. Otherwise, if the atom name starts with a C, N, O, S, or H, the element is assigned to the respective element. And otherwise otherwise, the first two characters of the atom name are used to search the upper()'d element symbols list. If that fails, then they try only the first character. If _that_ fails, then they return an empty string.

Reply:
Anyways, since the array data structure fed into the Molecule class has an array.element class variable, I don't need to wait until the blender object is created to get the element strings to then map to masses. I've repositioned the CoM chunk of code to the suggested place.